src/System.Collections.Immutable/src/Properties/SecurityTransparent.cs

```diff
@@ -0,0 +1,6 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Security;
+
+[assembly: SecurityTransparent]
```

Comment:
Supporting partial trust is not as simple as simple as adding a security transparent annotation.

You have to also make sure that the assembly actually works in partial trust, e.g. that  the security transparent code is not calling security critical code - both statically using [secanotate tool](http://blogs.msdn.com/b/shawnfa/archive/2009/11/18/using-secannotate-to-analyze-your-assemblies-for-transparency-violations-an-example.aspx), or dynamically (by running tests in partial trust environment).

Paying for this extra burden would be prohibitive across the board for corefx, paying for it in a few special cases should be ok.